test/net-hostlist-test.js

```diff
@@ -1032,6 +1063,429 @@ describe('Net HostList', function() {
     const entry = hosts.map.get(addr.hostname);
     assert(entry.addr.time > oldTime);
   });
+
+  it('should remove host', () => {
+    const hosts = new HostList();
+
+    const src = getRandomNetAddr();
+    const addrs = [
+      getRandomNetAddr(),
+      getRandomNetAddr(),
+      getRandomNetAddr(),
+      getRandomNetAddr()
+    ];
+
+    const used = addrs.slice(2);
+
+    for (const addr of addrs)
+      hosts.add(addr, src);
+
+    for (const addr of used)
+      hosts.markAck(addr.hostname, 0);
+
+    assert.strictEqual(hosts.map.size, addrs.length);
+    assert.strictEqual(hosts.totalUsed, 2);
+    assert.strictEqual(hosts.totalFresh, 2);
+    const fresh = getFreshEntries(hosts);
+    assert.strictEqual(fresh.length, 2);
+
+    assert.strictEqual(hosts.remove(getRandomIPv6()), null);
+    for (const addr of addrs.reverse())
+      assert.strictEqual(hosts.remove(addr.hostname), addr);
+    assert.strictEqual(hosts.totalUsed, 0);
+    assert.strictEqual(hosts.totalFresh, 0);
+  });
+
+  it('should mark ack', () => {
+    // we don't have the entry
+    {
+      const hosts = new HostList();
+      const addr = getRandomIPv4();
+      hosts.markAck(addr);
+    }
```

Comment:
Unfortunately this can not be tested, because we don't return anything useful from the markAck. So this just marks if (!entry) return; happened. Otherwise it would throw entry.addr.services |= services would fail.